src/eval.c

```diff
@@ -285,6 +285,7 @@ void scriptingInit(int setup) {
 void freeLuaScriptsSync(dict *lua_scripts, list *lua_scripts_lru_list, lua_State *lua) {
     dictRelease(lua_scripts);
     listRelease(lua_scripts_lru_list);
+    lua_gc(lctx.lua, LUA_GCCOLLECT, 0);
```

Comment:
Do we need to perform a full garbage collection cycle? Should we just stop the GC via `LUA_GCSTOP` ?

Reply:
I'll be honest, I do not understand the fix. I have been unable to reproduce the issue, and the CVE was not published yet so I don't know?

Reply:
Doing a `LUA_GCCOLLECT` should also be correct, however there could be a slight latency spike with that.

Reply:
> Doing a LUA_GCCOLLECT should also be correct, however there could be a slight latency spike with that.

Yes. We applied this fix internally already because it was disclosed previously, and haven't seen much issue as of yet.